docs/spl-docs/evaluation-functions/conversion-functions.md

```diff
@@ -58,31 +72,74 @@ If the `tonumber` function cannot parse a literal string to a number, it returns
+- The `stats` command then calculates the average of these numeric values (avgLoadTime), providing the development team with the information needed to identify performance bottlenecks and improve the user experience.
 
-If the input value is a number, it reformats it as a string. If the input value is Boolean, it returns the corresponding string value, "True" or "False".
+## **tostring(&lt;value&gt;, &lt;format&gt;)**
+
+This function converts a numeric, boolean, or NULL value to a string, optionally formatting the output based on the specified format. If the value is a field name and a row has NULL, it returns an empty string. The function does not directly support multivalue fields without preprocessing.
 
 ### Usage
 
-You can use this function with the `eval` and `where` commands and as part of evaluation expressions with other commands.
+The `tostring` function is used within `eval`, `where`, and other commands for converting numbers or boolean values to strings. The `<format>` argument is optional and can be used to apply specific formatting styles such as hexadecimal ("hex"), with commas ("commas"), or as a duration ("duration").
+
+### Formatting Details
 
-If `<value>` is a number, the second argument `<format>` is optional and can be "hex", "commas", or "duration".
+- **Hexadecimal ("hex")**: Converts a numeric value to its hexadecimal representation. If the value is not numeric, the result is undefined.
+- **With Commas ("commas")**: Formats a numeric value with commas as thousands separators. If the value is not numeric, the result is undefined.
+- **Duration ("duration")**: Converts a numeric value representing seconds into a duration format (e.g., "HH:MM:SS"). If the value is not numeric, the result is undefined.
 
 ### Example
 
-1. Converting 615 seconds to minutes and seconds, returning `period=615` and `period_time=00:10:15`:
+1. **Hexadecimal Conversion**: Converting a decimal number to a hexadecimal string:
+   ```spl
+   ... | eval hexValue=tostring(255, "hex")
+   ```
+   This converts the decimal number `255` into the hexadecimal string `"ff"`.
+
+2. **Formatting with Commas**: Converting a large number to a string with commas:
+   ```spl
+   ... | eval formattedNumber=tostring(1234567, "commas")
+   ```
+   This converts the number `1234567` into the string `"1,234,567"`.
+
+3. **Duration Formatting**: Converting seconds to a duration format:
+   ```spl
+   ... | eval duration=tostring(3661, "duration")
+   ```
+   This converts `3661` seconds into the string representing the duration `"1:01:01"` (1 hour, 1 minute, and 1 second).
+
+### Use-Case Example
+
+**Problem**: A financial application displays transaction amounts in a user's activity feed. To enhance readability, the application needs to format large transaction amounts with commas.
+
+**Solution**: Use the `tostring` function to convert numeric transaction amounts into formatted strings with commas.
+
```

Comment:
nit: are you saying they'd use `tostring` to format it and show it to the user? It would generally only be employees who are using Splunk from logs that were generated from their program running, so I don't think showing Splunk results to a user would make sense